src/modules/cmdpal/extensionsdk/Microsoft.CommandPalette.Extensions.Toolkit/ListSection.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.CommandPalette.Extensions.Toolkit;
+
+public partial class ListSection
+{
+    public virtual IListItem[] Items { get; set; } = [];
+
+    public virtual string SectionTitle { get; set; } = string.Empty;
+
+    private ListItem SectionListItem => new ListItem
```

Comment:
[Nit] Property that returns a new instance every time is a pure evil

```suggestion
    

    private ListItem CreateSectionListItem()
    {
        return new ListItem
        {
            Section = SectionTitle,
            Command = null,
        };
    }
```